### ISSUE TITLE
Long strings leave artifacts behind on To Scale Screen

### ISSUE THREAD

**Issue by loganbraywork** [phetsims/gravity-and-orbits#358]
**Test device**

Windows 10 Laptop

**Operating System**

Windows 10

**Browser**

Chrome

**Problem description**

For https://github.com/phetsims/QA/issues/519

In the To Scale screen, long strings will cause artifacts in the form of gray pixels to be left behind the various object descriptions as they travel. The pixels are hard to see in the included visual in the embedded size, but putting the gif in fullscreen seems to fix it. 

**Steps to reproduce**

1. Add the long query parameter
2. Enter the To Scale screen
3. Select the any model group 

**Visuals**

![2020-09-16Gravity OrbitsLongStringArtifacts](https://user-images.githubusercontent.com/54685448/93402694-15c9be00-f84b-11ea-9c92-0d97e61374d3.gif)

Troubleshooting information:
!!!!! DO NOT EDIT !!!!!
Name: ‪Gravity and Orbits‬:‪Gravity and Orbits‬
URL: https://phet-dev.colorado.edu/html/gravity-and-orbits/1.2.0-dev.16/phet/gravity-and-orbits_all_phet.html?stringTest=double
Version: 1.2.0-dev.16 2020-09-15 16:33:11 UTC
Features missing: generatedcontent, touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.105 Safari/537.36
Language: en-US
Window: 1536x754
Pixel Ratio: 1.25/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 30 uniform: 4096
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 32767x32767
OES_texture_float: true
Dependencies JSON: {}

**Comment by samreid** [comment]
@jonathanolson is the solution to this problem still to add an invisible rectangle of padding around the text, to increase its bounds?

**Comment by jonathanolson** [comment]
> @jonathanolson is the solution to this problem still to add an invisible rectangle of padding around the text, to increase its bounds?

That sounds good to try first!

**Comment by samreid** [comment]
I added a 1px-dilation rectangle underlay, can you please test if this corrects the problem?

**Comment by KatieWoe** [comment]
I am having trouble reproducing/seeing this in the version it was reported in, so I'll let @loganbraywork confirm it for rc.2

**Comment by loganbraywork** [comment]
This issue is still present and seems to be worse in the newest RC test, with an increase in artifacting when the planets are moving towards the bottom of the screen.  Also noting that the link in the original issue is incorrect, specifically the double query parameter does not cause the bug, only the long stringtest.
![2020-11-03GravityandOrbitsArtifactingRC2](https://user-images.githubusercontent.com/54685448/98051701-a8b9ba00-1df1-11eb-911d-ee46979954e0.gif)

**Comment by samreid** [comment]
It looks like the underlay is also reduced by the maxWidth, so maybe pixels are still leaking out:

![image](https://user-images.githubusercontent.com/679486/98429380-78795200-2063-11eb-81d5-c8860abe9a7b.png)

**Comment by samreid** [comment]
I increased the size of the underlay, @loganbraywork can you please test?

**Comment by loganbraywork** [comment]
I'm still seeing the issue on phettest

**Comment by samreid** [comment]
Even when I comment out the transparent-rectangle workaround, I am unable to see the problem when testing on Windows Chrome 86.0.4240.193.  @KatieWoe also reported she cannot see the problem in the affected version.  It could be that the platform is using a different font or has different font metrics, and Chrome doesn't know to clear that area.  Additionally, it could be that the transparent rectangle doesn't trigger that platform to repaint.

Since the standard solution is not working, and I have no way to reproduce the problem, let's discuss possible next steps.  

1. Close this issue as "will not fix"
2. I publish a new version where the overlay rectangles are not fully transparent, and we see if that corrects the problem on the device
3. We check with other PhET developers to see if they can reproduce the problem on their Win/Chrome
4. We set up a pairing session between me and @loganbraywork to experiment with different ideas (like (2) above, but in realtime).

@arouinfar or @ariel-phet can you please advise?

**Comment by arouinfar** [comment]
I'm leaning towards closing this as wont-fix, assuming the artifacts have returned to the more subtle variety shown in the original report.

@loganbraywork can you let us know what the artifacts currently look like in master? Are they still very obvious like in https://github.com/phetsims/gravity-and-orbits/issues/358#issuecomment-721428332 or are they more subtle like they were originally in https://github.com/phetsims/gravity-and-orbits/issues/358#issue-703145433.

**Comment by samreid** [comment]
Here is a dev version we can use to rule out caching issues, @loganbraywork can you please use it to comment on the current state of the artifacts?

https://phet-dev.colorado.edu/html/gravity-and-orbits/1.3.0-dev.1/phet/gravity-and-orbits_en_phet.html

**Comment by samreid** [comment]
I recommend we unblock this for the upcoming milestone.  The client is unlikely to use long strings.  Also, if we find a fix, we can maintenance patch it in.

**Comment by arouinfar** [comment]
Sounds good to me @samreid. I'll remove the blocks-sim-publication tag.

**Comment by ariel-phet** [comment]
@samreid @arouinfar as this issue is unlikely to come up in actual use, I am just going to go ahead and mark `wontfix` and close.

**Comment by loganbraywork** [comment]
I do still see the issue, although I am not sure why. I travelled home and tested in on a different laptop and had the same result.

**Comment by KatieWoe** [comment]
I started seeing this in text area on ChromeOS with no string tests. @samreid should we look at this?

**Comment by samreid** [comment]
Can you please clarify what you meant by "in text area"?

**Comment by KatieWoe** [comment]
The phet-io text area wrapper

**Comment by ariel-phet** [comment]
@samreid I am still not seeing this issue as necessary to fix as I doubt long strings will be introduced in these places, but let me know if you want to discuss.

**Comment by samreid** [comment]
I don't think it is critical to fix this for our upcoming release, but it would be nice to understand why our standard solution didn't work in this case--something puzzling is going on.  I recommend we leave this open a bit longer to see if I can reproduce it on a chromebook.  If not, we will have to decide whether it is worth the time to ask for another developer try to reproduce the problem, or have me pair with a QA team member.  Sound OK?

**Comment by ariel-phet** [comment]
@samreid sounds great. Happy to leave open until you solve the mystery or decide no longer worth investigation.

**Comment by samreid** [comment]
Testing on a Lenovo N23 Yoga Chromebook running Chrome 86.0.4240.199, I was able to see the problem in the data: text area wrapper without any query parameters.  I found that `new Rectangle()` with no fill did not trigger a repaint, but that `new Rectangle({fill:'blue',opacity:0})` did trigger a repaint.  I committed that to the 1.3 branch and to master.  @KatieWoe or @loganbraywork can you please test after pull-all on https://bayes.colorado.edu/dev/phettest/ ?  Close if the problem is no longer replicable.

I'll open a new issue about whether this problem impacts other sims.

**Comment by KatieWoe** [comment]
Did not see it on master in the case I encountered. @loganbraywork can you check the case that you've been seeing?

**Comment by loganbraywork** [comment]
Looks good to me as of the Latest RC Test https://github.com/phetsims/QA/issues/595
